### PR TITLE
Added documentation for changing the default values of HNSW hyper parameters in k-NN plugin

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Added documentation for changing the default values of HNSW hyper parameters in k-NN plugin

### Issues Resolved
Resolves https://github.com/opensearch-project/documentation-website/issues/6021

k-NN plugin issue: https://github.com/opensearch-project/k-NN/issues/1354

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @navneet1v ! Is this PR ready for doc review?

**Comment by navneet1v** [comment]
@hdhalter yes it is.

File: _search-plugins/knn/knn-index.md [hunk 1]
```diff
@@ -39,12 +39,15 @@ Method name | Requires training | Supported spaces | Description
...
 Parameter name | Required | Default | Updatable | Description
 :--- | :--- | :--- | :--- | :---
-`ef_construction` | false | 512 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
+`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
 `m` | false | 16 | false | The number of bidirectional links that the plugin creates for each new element. Increasing and decreasing this value can have a large impact on memory consumption. Keep this value between 2 and 100.
 
 For nmslib, *ef_search* is set in the [index settings](#index-settings).
 {: .note}
 
+Index created with version older than 2.12 of Opensearch will still use the old value of `ef_construction` which is `512`
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-index.md:48]:
```suggestion
An index created in OpenSearch version 2.12 or earlier will still use the old `ef_construction` value (`512`).
```

File: _search-plugins/knn/knn-index.md [hunk 2]
```diff
@@ -59,11 +62,14 @@ For hnsw, "innerproduct" is not available when PQ is used.
...
-`ef_search` | false | 512 | false | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches.
-`ef_construction` | false | 512 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
+`ef_search` | false | 100 | false | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches.
+`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
 `m` | false | 16 | false | The number of bidirectional links that the plugin creates for each new element. Increasing and decreasing this value can have a large impact on memory consumption. Keep this value between 2 and 100.
 `encoder` | false | flat | false | Encoder definition for encoding vectors. Encoders can reduce the memory footprint of your index, at the expense of search accuracy.
 
+Index created with version older than 2.12 of Opensearch will still use the old value of `ef_construction` and `ef_search` which is `512`
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-index.md:70]:
```suggestion
An index created in OpenSearch version 2.12 or earlier will still use the old `ef_construction` and `ef_search` value (`512`).
```

File: _search-plugins/knn/knn-index.md [hunk 3]
```diff
@@ -91,12 +97,15 @@ Method name | Requires training | Supported spaces | Description
...
 `m` | false | 16 | false | The number of bidirectional links that the plugin creates for each new element. Increasing and decreasing this value can have a large impact on memory consumption. Keep this value between 2 and 100.<br>The Lucene engine uses the proprietary term "max_connections" to describe this function, which corresponds directly to "m". To be consistent throughout OpenSearch documentation, we retain the term "m" to label this parameter.
 
 Lucene HNSW implementation ignores `ef_search`  and dynamically sets it to the value of "k" in the search request. Therefore, there is no need to make settings for `ef_search` when using the Lucene engine.
 {: .note}
 
+Index created with version older than 2.12 of Opensearch will still use the old value of `ef_construction` which is `512`
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-index.md:106]:
```suggestion
An index created in OpenSearch version 2.12 or earlier will still use the old `ef_construction` value (`512`).
```

File: _search-plugins/knn/knn-index.md [hunk 4]
```diff
@@ -251,7 +260,10 @@ At the moment, several parameters defined in the settings are in the deprecation
...
-`index.knn.algo_param.ef_construction` | 512 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
+`index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches. Only available for nmslib.
+`index.knn.algo_param.ef_construction` | 100 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
 `index.knn.algo_param.m` | 16 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
 `index.knn.space_type` | l2 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
+
+Index created with version older than 2.12 of Opensearch will still use the old value of `ef_construction` and `ef_search` which is `512`
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-index.md:268]:
```suggestion
An index created in OpenSearch version 2.12 or earlier will still use the old `ef_construction` and `ef_search` value (`512`).
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @navneet1v! A couple of rewrites.

File: _search-plugins/knn/knn-index.md (hunk 3 above)
Comment (navneet1v) [reply _search-plugins/knn/knn-index.md:106]: ack

File: _search-plugins/knn/knn-index.md (hunk 2 above)
Comment (navneet1v) [reply _search-plugins/knn/knn-index.md:70]: ack

File: _search-plugins/knn/knn-index.md (hunk 1 above)
Comment (navneet1v) [reply _search-plugins/knn/knn-index.md:48]: ack

File: _search-plugins/knn/knn-index.md [hunk 5]
```diff
@@ -39,12 +39,15 @@ Method name | Requires training | Supported spaces | Description
 
 Parameter name | Required | Default | Updatable | Description
 :--- | :--- | :--- | :--- | :---
-`ef_construction` | false | 512 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
+`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:42]:
```suggestion
`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values result in a more accurate graph but slower indexing speed.
```

File: _search-plugins/knn/knn-index.md [hunk 6]
```diff
@@ -59,11 +62,14 @@ For hnsw, "innerproduct" is not available when PQ is used.
 
 Parameter name | Required | Default | Updatable | Description
 :--- | :--- | :--- | :--- | :---
-`ef_search` | false | 512 | false | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches.
-`ef_construction` | false | 512 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
+`ef_search` | false | 100 | false | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:65]:
```suggestion
`ef_search` | false | 100 | false | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches.
```

File: _search-plugins/knn/knn-index.md [hunk 7]
```diff
@@ -59,11 +62,14 @@ For hnsw, "innerproduct" is not available when PQ is used.
...
+`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:66]:
```suggestion
`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values result in a more accurate graph but slower indexing speed.
```

File: _search-plugins/knn/knn-index.md [hunk 8]
```diff
@@ -59,11 +62,14 @@ For hnsw, "innerproduct" is not available when PQ is used.
...
 `m` | false | 16 | false | The number of bidirectional links that the plugin creates for each new element. Increasing and decreasing this value can have a large impact on memory consumption. Keep this value between 2 and 100.
 `encoder` | false | flat | false | Encoder definition for encoding vectors. Encoders can reduce the memory footprint of your index, at the expense of search accuracy.
 
+An index created in OpenSearch version 2.11 or earlier will still use the old `ef_construction` and `ef_search` value (`512`).
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:70]: "values" (plural)?

File: _search-plugins/knn/knn-index.md [hunk 9]
```diff
@@ -91,12 +97,15 @@ Method name | Requires training | Supported spaces | Description
 
 Parameter name | Required | Default | Updatable | Description
 :--- | :--- | :--- | :--- | :---
-`ef_construction` | false | 512 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.<br>The Lucene engine uses the proprietary term "beam_width" to describe this function, which corresponds directly to "ef_construction". To be consistent throughout OpenSearch documentation, we retain the term "ef_construction" to label this parameter.
+`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed.<br>The Lucene engine uses the proprietary term "beam_width" to describe this function, which corresponds directly to "ef_construction". To be consistent throughout OpenSearch documentation, we retain the term "ef_construction" to label this parameter.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:100]:
```suggestion
`ef_construction` | false | 100 | false | The size of the dynamic list used during k-NN graph creation. Higher values result in a more accurate graph but slower indexing speed.<br>The Lucene engine uses the proprietary term "beam_width" to describe this function, which corresponds directly to "ef_construction". To be consistent throughout the OpenSearch documentation, we retain the term "ef_construction" for this parameter.
```

File: _search-plugins/knn/knn-index.md [hunk 10]
```diff
@@ -251,7 +260,10 @@ At the moment, several parameters defined in the settings are in the deprecation
 Setting | Default | Updatable | Description
 :--- | :--- | :--- | :---
 `index.knn` | false | false | Whether the index should build native library indexes for the `knn_vector` fields. If set to false, the `knn_vector` fields will be stored in doc values, but Approximate k-NN search functionality will be disabled.
-`index.knn.algo_param.ef_search` | 512 | true | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches. Only available for nmslib.
-`index.knn.algo_param.ef_construction` | 512 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
+`index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches. Only available for nmslib.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:263]:
```suggestion
`index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches. Only available for nmslib.
```

Comment (natebower) [inline _search-plugins/knn/knn-index.md:263]: Should it be NMSLIB or `nmslib`?

File: _search-plugins/knn/knn-index.md [hunk 11]
```diff
@@ -251,7 +260,10 @@ At the moment, several parameters defined in the settings are in the deprecation
...
+`index.knn.algo_param.ef_construction` | 100 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:264]:
```suggestion
`index.knn.algo_param.ef_construction` | 100 | false | Deprecated in 1.0.0. Instead, use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value.
```

File: _search-plugins/knn/knn-index.md [hunk 12]
```diff
@@ -251,7 +260,10 @@ At the moment, several parameters defined in the settings are in the deprecation
...
 `index.knn.algo_param.m` | 16 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
 `index.knn.space_type` | l2 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
+
+An index created in OpenSearch version 2.11 or earlier will still use the old `ef_construction` and `ef_search` value (`512`).
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:268]: "values" (plural)?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/knn/knn-index.md (hunk 10 above)
Comment (navneet1v) [reply _search-plugins/knn/knn-index.md:263]: it should be nmslib only.

**Comment by navneet1v** [comment]
@natebower I fixed all your comments. Please review.

File: _search-plugins/knn/knn-index.md [hunk 13]
```diff
@@ -251,7 +260,10 @@ At the moment, several parameters defined in the settings are in the deprecation
 Setting | Default | Updatable | Description
 :--- | :--- | :--- | :---
 `index.knn` | false | false | Whether the index should build native library indexes for the `knn_vector` fields. If set to false, the `knn_vector` fields will be stored in doc values, but Approximate k-NN search functionality will be disabled.
-`index.knn.algo_param.ef_search` | 512 | true | The size of the dynamic list used during k-NN searches. Higher values lead to more accurate but slower searches. Only available for nmslib.
-`index.knn.algo_param.ef_construction` | 512 | false | Deprecated in 1.0.0. Use the [mapping parameters](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions) to set this value instead.
+`index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches. Only available for nmslib.
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:263]:
```suggestion
`index.knn.algo_param.ef_search` | 100 | true | The size of the dynamic list used during k-NN searches. Higher values result in more accurate but slower searches. Only available for NMSLIB.
```

**Review by natebower — APPROVED** [review]
@navneet1v @kolchfa-aws Just one change. Otherwise, LGTM.

**Pull request merged**